src/components/user/user.service.ts

```diff
@@ -285,17 +284,78 @@ export class UserService {
     return result.id;
   }
 
-  async readOne(
-    id: string,
-    { userId }: Pick<Session, 'userId'>
-  ): Promise<User> {
+  async readOne(id: string, session: Session): Promise<User> {
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'User', { id })])
+      .call(matchPropListNew)
+      .return('propList, node')
+      .asResult<StandardReadResult<DbPropsOfDto<User>>>();
+
+    const result = await query.first();
+    if (!result) {
+      throw new NotFoundException('Could not find user', 'user.id');
+    }
+
+    const rolesValue = result.propList
+      .filter((prop) => prop.property === 'roles')
+      .map((prop) => prop.value as Role);
+
+    let permsOfBaseNode = {};
+    // -- let the user explicitly see all properties only if they're reading their own ID
+    // -- TODO: need to find a more... dynamic way of doing this. Just coding like this for
+    //    now to make test pass and this isn't a huge priority at the moment.
+    if (id === session.userId) {
+      permsOfBaseNode = {
+        about: { canRead: true, canEdit: true },
+        displayFirstName: { canRead: true, canEdit: true },
+        displayLastName: { canRead: true, canEdit: true },
+        email: { canRead: true, canEdit: true },
+        phone: { canRead: true, canEdit: true },
+        realFirstName: { canRead: true, canEdit: true },
+        realLastName: { canRead: true, canEdit: true },
+        roles: { canRead: true, canEdit: true },
+        status: { canRead: true, canEdit: true },
+        timezone: { canRead: true, canEdit: true },
+        title: { canRead: true, canEdit: true },
+        education: { canRead: true, canEdit: true },
+        organization: { canRead: true, canEdit: true },
+        unavailability: { canRead: true, canEdit: true },
+        locations: { canRead: true, canEdit: true },
+        knownLanguage: { canRead: true, canEdit: true },
+      };
+    } else {
+      permsOfBaseNode = await this.authorizationService.getPermissionsOfBaseNode(
+        new DbUser(),
+        session
+      );
+    }
+
+    const securedProps = parseSecuredPropertiesNew(
+      result.propList,
+      this.securedProperties,
+      permsOfBaseNode
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
+      roles: {
+        ...securedProps.roles,
+        value: rolesValue,
+      },
+      canDelete: await this.db.checkDeletePermission(id, session),
+    };
+  }
+
+  async readOneByUserId(id: string, userId: string): Promise<User> {
```

Comment:
See comment on projects about merging these two methods.

Reply:
see above reply... did pretty much the same thing with user as with project